ml-algorithms/src/main/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactory.java

```diff
@@ -8,79 +8,63 @@
 
 @Log4j2
 public class MLHttpClientFactory {
 
-    public static CloseableHttpClient getCloseableHttpClient() {
-        return createHttpClient();
+    public static SdkAsyncHttpClient getAsyncHttpClient() {
+        try {
+            return AccessController
+                .doPrivileged(
+                    (PrivilegedExceptionAction<SdkAsyncHttpClient>) () -> NettyNioAsyncHttpClient.builder().maxConcurrency(100).build()
+                );
+        } catch (PrivilegedActionException e) {
+            return null;
+        }
     }
 
-    private static CloseableHttpClient createHttpClient() {
-        HttpClientBuilder builder = HttpClientBuilder.create();
-
-        // Only allow HTTP and HTTPS schemes
-        builder.setSchemePortResolver(new DefaultSchemePortResolver() {
-            @Override
-            public int resolve(HttpHost host) throws UnsupportedSchemeException {
-                validateSchemaAndPort(host);
-                return super.resolve(host);
-            }
-        });
-
-        builder.setDnsResolver(MLHttpClientFactory::validateIp);
-
-        builder.setRedirectStrategy(new LaxRedirectStrategy() {
-            @Override
-            public boolean isRedirected(HttpRequest request, HttpResponse response, HttpContext context) {
-                // Do not follow redirects
-                return false;
+    public static void validate(String protocol, String host, String port) throws UnknownHostException {
+        if (protocol != null && !"http".equalsIgnoreCase(protocol) && !"https".equals(protocol)) {
+            log.error("Remote inference protocol is not http or https: " + protocol);
+            throw new IllegalArgumentException("Protocol is not http or https: " + protocol);
+        }
+        String portStr = Optional.ofNullable(port).orElseGet(() -> {
+            if (protocol == null || "http".equals(protocol.toLowerCase(Locale.getDefault()))) {
+                return "80";
+            } else {
+                return "443";
             }
         });
-        return builder.build();
-    }
-
-    @VisibleForTesting
-    protected static void validateSchemaAndPort(HttpHost host) {
-        String scheme = host.getSchemeName();
-        if ("http".equalsIgnoreCase(scheme) || "https".equalsIgnoreCase(scheme)) {
-            String[] hostNamePort = host.getHostName().split(":");
-            if (hostNamePort.length > 1 && NumberUtils.isDigits(hostNamePort[1])) {
-                int port = Integer.parseInt(hostNamePort[1]);
-                if (port < 0 || port > 65536) {
-                    log.error("Remote inference port out of range: " + port);
-                    throw new IllegalArgumentException("Port out of range: " + port);
-                }
-            }
-        } else {
-            log.error("Remote inference scheme not supported: " + scheme);
-            throw new IllegalArgumentException("Unsupported scheme: " + scheme);
+        if (!NumberUtils.isDigits(portStr)) {
```

Comment:
why can't we return `80` or `443` in the first place? Why do we need to return as string?

Reply:
Changed this as the protocol, host and port retrieval approach changed.